test/model/sequence_result_test.rb

```diff
@@ -3,11 +3,310 @@
 require_relative '../test_helper'
 require_relative '../../lib/app/models/sequence_result'
 require_relative '../shared/result_status_tests'
+require_relative '../../lib/app/utils/result_statuses'
 
 class SequenceResultTest < MiniTest::Test
   include Inferno::ResultStatusTests
 
   def setup
     @result = Inferno::Models::SequenceResult.new
   end
+
+  def test_reset!
+    result = Inferno::Models::SequenceResult.new
+    result.required_passed = 20
+    result.required_total = 15
+    result.error_count = 2
+    result.todo_count = 3
+    result.skip_count = 6
+    result.optional_passed = 2
+    result.optional_total = 3
+    result.required_omitted = 4
+    result.optional_omitted = 300
+
+    result.reset!
+
+    assert_equal 0, result.required_passed
+    assert_equal 0, result.required_total
+    assert_equal 0, result.error_count
+    assert_equal 0, result.todo_count
+    assert_equal 0, result.skip_count
+    assert_equal 0, result.optional_passed
+    assert_equal 0, result.optional_total
+    assert_equal 0, result.required_omitted
+    assert_equal 0, result.optional_omitted
+  end
+
+  def test_result_count
+    result = Inferno::Models::SequenceResult.new
+    result.test_results << Inferno::Models::TestResult.new
+    result.test_results << Inferno::Models::TestResult.new
+
+    assert_equal 2, result.result_count
+  end
+
+  def test_total_omitted
+    result = Inferno::Models::SequenceResult.new
+
+    result.required_omitted = 4
+    result.optional_omitted = 300
+
+    assert_equal 304, result.total_omitted
+  end
+
+  def test_update_result_counts
+    sequence_result = Inferno::Models::SequenceResult.new
+
+    test_result1 = Inferno::Models::TestResult.new
```

Comment:
Could you DRY up these tests? Perhaps something like:

```
result_params = [
  { required: true, result: Inferno::ResultStatuses::PASS },
  { required: false, result: Inferno::ResultStatuses::PASS },
   ...
]
sequence_result.test_results = result_params.map { |params| Inferno::Models::TestResult.new(params) }
```

Reply:
Updated tests to follow this format